lib/node_modules/@stdlib/lapack/base/dlaswp/lib/dlaswp.js

```diff
@@ -0,0 +1,144 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isLayout = require( '@stdlib/blas/base/assert/is-layout' );
+var floor = require( '@stdlib/math/base/special/floor' );
+var dswap = require( '@stdlib/blas/base/dswap' ).ndarray;
+
+
+// MAIN //
+
+/**
+* Performs a series of row interchanges on the matrix `A` using pivot indices stored in `IPIV`.
+*
+* @param {string} order - storage layout
+* @param {PositiveInteger} N - number of columns of `A`
+* @param {Float64Array} A - matrix
+* @param {PositiveInteger} LDA - leading dimension of `A`
+* @param {PositiveInteger} k1 - index of first row to interchange
+* @param {PositiveInteger} k2 - index of last row to interchange
+* @param {Int32Array} IPIV - vector of pivot indices
+* @param {PositiveInteger} incX - increment between successive values of `IPIV`
+* @throws {TypeError} first argument must be a valid order
+* @returns {Float64Array} permuted matrix `A`
+*
+* @example
+* var Int32Array = require( '@stdlib/array/int32' );
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var IPIV = new Int32Array( [ 2, 0, 1 ] );
+* var A = new Float64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+*
+* dlaswp( 'column-major', 3, A, 2, 0, 2, IPIV, 1 );
+* // A => <Float64Array>[ 2.0, 3.0, 4.0, 5.0, 1.0, 6.0 ]
+*/
+function dlaswp( order, N, A, LDA, k1, k2, IPIV, incX ) {
```

Comment:
```suggestion
function dlaswp( order, N, A, LDA, k1, k2, IPIV, sx ) {
```

I think we need to rename this variable to follow conventions, I am not exactly sure to what we shall.

Reply:
Use `strideX`. If we are being pedantic, this should be `strideIPIV`. I am fine with the latter in order to avoid ambiguity, since the signature doesn't have an `X` parameter.

Reply:
I agree with strideIPIV.